WordPress/gutenberg
# Lodash: Refactor away from `_.isPlainObject()`

## What?
This PR removes the `_.isPlainObject()` usage completely and deprecates the function. 

Alternative to #42471. 

Part of #16938.

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

We're using the [is-plain-obj](https://github.com/sindresorhus/is-plain-obj) package because it does exactly what we want, has type bindings, and is simple and well-contained enough.

## Testing Instructions
* Verify all tests still pass.
* Smoke test all editors and verify things look good.
